src/index-worker.ts

```diff
@@ -1,39 +1,69 @@
-import { IpcMessageEvent, ipcRenderer, remote} from "electron";
+import { IpcMessageEvent, ipcRenderer, remote } from "electron";
+import _ = require("lodash");
+import net = require("net");
 import { basename } from "path";
 import { Repository } from "./common/repository";
-import { initExceptionManager } from "./exceptionManager";
+import { initExceptionManager, notify } from "./exceptionManager";
 import { repStore } from "./repoStore";
 import * as graphQlServer from "./server";
 
 let mainWindowId = -1;
 
 const getRepos = () => repStore.getRepositories().map((r) => r.toModel());
 
+const isPortInUse = (port: number): Promise<boolean> => new Promise<boolean>((resolve, reject) => {
+    const testServer = net.createServer()
+    .on("error", (err: any) => {
+        err.code === "EADDRINUSE" ? resolve(true) : reject(err);
+    })
+    .on("listening", () => {
+        testServer.once("close", () => resolve(false));
```

Comment:
What happens if you cant close the server? You're doing the same mistake they did

Reply:
What mistake? who did it?  
If the `'listening'` event is triggered, the server is running and can be closed.  
If there is an error with the close operation - the error event will trigger